src/sentry/plugins/providers/integration_repository.py

```diff
@@ -96,12 +102,34 @@ def dispatch(self, request: Request, organization, **kwargs):
                     self.on_delete_repository(repo)
                 except IntegrationError:
                     pass
-                return Response(
-                    {"errors": {"__all__": "A repository with that name already exists"}},
-                    status=400,
-                )
+
+                if is_dispatch:
+                    return (
+                        Response(
+                            {"errors": {"__all__": "A repository with that name already exists"}},
+                            status=400,
+                        ),
+                        None,
```

Comment:
it's difficult to isolate the logic of creating a repo row when returned API responses are nested within the original logic from the `dispatch` function, so i included a boolean here. i'm not a fan of doing this so any other ideas would be appreciated